src/main/java/com/linkedin/kmf/services/ConsumerFactoryImpl.java

```diff
@@ -81,10 +81,27 @@ public ConsumerFactoryImpl(Map<String, Object> props) throws Exception {
       props.forEach(consumerProps::putIfAbsent);
     }
 
-    _baseConsumer = (KMBaseConsumer) Class.forName(consumerClassName)
-        .getConstructor(String.class, Properties.class, AdminClient.class)
-        .newInstance(_topic, consumerProps, adminClient());
+    if (adminClientConstructorExists(consumerClassName)) {
+      _baseConsumer = (KMBaseConsumer) Class.forName(consumerClassName)
+          .getConstructor(String.class, Properties.class, AdminClient.class)
```

Comment:
Rather than reflect twice (i.e., the new helper and here), maybe you can arrange to have the helper return the constructor if it exists, null otherwise?